packages/fuel-gauge/src/contract.test.ts

```diff
@@ -946,4 +946,24 @@ describe('Contract', () => {
 
     expect(finalBalance).toBe(initialBalance + amountToContract);
   });
+
+  it('should ensure ScriptResultDecoderError works for dryRun and simulate calls', async () => {
```

Comment:
Says `simulate` but we only seem to `dryRun` in the test case. I appreciate these are coupled at the moment but they could be decoupled at some point, is it worth having a separate test case for `simulate`?